--- a/CMakeLists.txt	2023-10-12 11:02:44
+++ b/CMakeLists.txt	2023-10-12 11:03:11
@@ -15,6 +15,8 @@
 
 set( CMAKE_DIRECTORY_LABELS "fv3-jedi-lm" )
 
+find_package( ecbuild 3.6 REQUIRED HINTS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/../ecbuild)
+
 set( ECBUILD_DEFAULT_BUILD_TYPE Release )
 set( ENABLE_OS_TESTS           OFF CACHE BOOL "Disable OS tests" FORCE )
 set( ENABLE_LARGE_FILE_SUPPORT OFF CACHE BOOL "Disable testing of large file support" FORCE )
