diff --git a/CMakeLists.txt b/CMakeLists.txt
index 699d1f90..009117ac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,7 +92,7 @@ macro(mpas_fortran_target _tgt)
     # Fortran modules include path
     set_target_properties(${_tgt} PROPERTIES Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/${MPAS_MODULE_DIR})
     target_include_directories(${_tgt} INTERFACE $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/${MPAS_MODULE_DIR}>
-                                                 $<INSTALL_INTERFACE:${MPAS_MODULE_DIR}>)
+                                                 $<INSTALL_INTERFACE:${CMAKE_INSTALL_LIBDIR}/${MPAS_MODULE_DIR}>)
     #Relocatable, portable, runtime dynamic linking
     set_target_properties(${_tgt} PROPERTIES INSTALL_RPATH "\$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
 
@@ -161,7 +161,7 @@ macro(mpas_core_target)
     set(CORE_MODULE_DIR ${MPAS_MODULE_DIR}/${_ARG_TARGET})
     set_target_properties(${_ARG_TARGET} PROPERTIES Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/${CORE_MODULE_DIR})
     target_include_directories(${_ARG_TARGET} INTERFACE $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/${CORE_MODULE_DIR}>
-                                                        $<INSTALL_INTERFACE:${CORE_MODULE_DIR}>)
+                                                        $<INSTALL_INTERFACE:${CMAKE_INSTALL_LIBDIR}/${CORE_MODULE_DIR}>)
 
     #MPAS Specific option
     target_compile_definitions(${_ARG_TARGET} PRIVATE ${_TGT}=1)
@@ -300,7 +300,7 @@ install(EXPORT ${PROJECT_NAME}ExportsExternal NAMESPACE ${PROJECT_NAME}::externa
         FILE ${_project_name_lower}-targets-external.cmake
         DESTINATION ${CONFIG_INSTALL_DESTINATION})
 install(EXPORT ${PROJECT_NAME}Exports NAMESPACE ${PROJECT_NAME}::
-        FILE ${_project_name_lower}-targets.cmake
+        FILE ${_project_name_lower}-targets-main.cmake
         DESTINATION ${CONFIG_INSTALL_DESTINATION})
 install(EXPORT ${PROJECT_NAME}ExportsCore NAMESPACE ${PROJECT_NAME}::core::
         FILE ${_project_name_lower}-targets-core.cmake
diff --git a/cmake/PackageConfig.cmake.in b/cmake/PackageConfig.cmake.in
index 424e9ba8..5a83779f 100644
--- a/cmake/PackageConfig.cmake.in
+++ b/cmake/PackageConfig.cmake.in
@@ -35,11 +35,13 @@
 
 #  * @PROJECT_NAME@::@PROJECT_NAME@_shared - shared library target:
 
-#Include targets file.  This will create IMPORTED target @PROJECT_NAME@
+#Include targets file.  This will create IMPORTED target @PROJECT_NAME@.
+# Note that some of these targets import each other, so establishing
+# the correct include order is critical.
 string(TOLOWER @PROJECT_NAME@ _project_name_lower)
 if(NOT TARGET @PROJECT_NAME@::framework)
     include("${CMAKE_CURRENT_LIST_DIR}/${_project_name_lower}-targets-external.cmake")
-    include("${CMAKE_CURRENT_LIST_DIR}/${_project_name_lower}-targets.cmake")
+    include("${CMAKE_CURRENT_LIST_DIR}/${_project_name_lower}-targets-main.cmake")
     include("${CMAKE_CURRENT_LIST_DIR}/${_project_name_lower}-targets-core.cmake")
 endif()
 
